src/Microsoft.Framework.Configuration.FileExtensions/FileConfigurationExtensions.cs

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+
+namespace Microsoft.Framework.Configuration
+{
+    public static class FileConfigurationExtensions
+    {
+        /// <summary>
+        /// Sets the base path in <see cref="IConfigurationBuilder"/> for file based providers.
+        /// </summary>
+        /// <param name="builder">The <see cref="IConfigurationBuilder"/> to add to.</param>
+        /// <param name="basePath">Absolute path of file based providers.
+        /// <returns>The <see cref="IConfigurationBuilder"/>.</returns>
+        public static IConfigurationBuilder SetBasePath(this IConfigurationBuilder builder, string basePath)
+        {
+            if (builder == null)
+            {
+                throw new ArgumentNullException(nameof(builder));
+            }
+
+            if (basePath == null)
+            {
+                throw new ArgumentNullException(nameof(basePath));
+            }
+
+            builder.Properties["BasePath"] = basePath;
+            
+            return builder;
+        }
+
+        /// <summary>
+        /// Gets the base path in <see cref="IConfigurationBuilder"/> for file based providers.
+        /// </summary>
+        /// <param name="builder">The <see cref="IConfigurationBuilder"/> to add to.</param>
+        /// <returns>The <see cref="IConfigurationBuilder"/>.</returns>
+        public static string GetBasePath(this IConfigurationBuilder builder)
+        {
+            if (builder == null)
+            {
+                throw new ArgumentNullException(nameof(builder));
+            }
+
+            object basePath;
+            if (builder.Properties.TryGetValue("BasePath", out basePath))
+            {
+                return (string)basePath;
+            }
+
+            return null;
+        }
+
+        public static string GetConfigurationFilePath(this IConfigurationBuilder configuration, string path)
```

Comment:
Moreover, it is now clear that this method is just plain vanilla `Path.Combine(configurationBuilder.GetBasePath(), filePath)`. I am thinking we don't really need it anymore. Even the logic for #295 will go inside `GetBasePath()`. Can we get rid of it?